### PR TITLE
Avoid multiplayer spawner crash.

### LINKED ISSUE #61390: MultiplayerSpawner.new().get_spawnable_scene(100) crash
### Godot version

4dd6f56c82fd87b85ad483e6f95585a83cd88a12

### System information

popos 22.04 Linux, Vulkan, NVIDIA 3000 series

### Issue description

```
MultiplayerSpawner.get_spawnable_scene --- executing with 1 parameters [100]
GDSCRIPT CODE:     MultiplayerSpawner.new().get_spawnable_scene(100)
ERROR: Index p_index = 100 is out of bounds (count = 0).
   at: operator[] (./core/templates/local_vector.h:161)
```

### Steps to reproduce

Run https://github.com/godotengine/regression-test-project from vscode.

```json
  {
    "name": "Launch regression test",
    "type": "lldb",
    "request": "launch",
    "cwd": "${workspaceFolder}",
    "miDebuggerPath": "/usr/bin/gdb",
    "args": ["--audio-driver", "Dummy", "--path", "regression-tests"],
    "program": "${workspaceFolder}/godot/bin/godot.linuxbsd.opt.tools.64.llvm",
    "visualizerFile": "${workspaceFolder}/godot/platform/windows/godot.natvis",
    "preLaunchTask": "Build Linux"
  },
```

```
{
  "label": "Build Linux",
  "type": "shell",
  "command": "scons target=release_debug compiledb=yes float=32 dev=yes werror=yes platform=linuxbsd target=release_debug -j`nproc` use_lto=no deprecated=no use_static_cpp=yes use_llvm=yes builtin_freetype=yes custom_modules=../godot-modules-groups",
  "options": {
    "cwd": "${workspaceFolder}/godot"
  },
  "problemMatcher": [],
  "group": {
    "kind": "build",
    "isDefault": true
  }
},
```

### Minimal reproduction project

https://github.com/godotengine/regression-test-project

### PR DESCRIPTION
**Pull request by fire**
Fixes: https://github.com/godotengine/godot/issues/61390

In the second function, we don't handle the case where local vector `spawnable_scenes[p_idx]` is indexed with the wrong index.

Since apparently it doesn't crash via the test program.

### CODE REVIEW

**Comment by fire** [comment]
LocalVector is flaky.

File: scene/multiplayer/multiplayer_spawner.cpp [hunk 1]
```diff
@@ -97,12 +98,21 @@ void MultiplayerSpawner::add_spawnable_scene(const String &p_path) {
 	}
 	spawnable_scenes.push_back(sc);
 }
+
 int MultiplayerSpawner::get_spawnable_scene_count() const {
+	if (spawnable_scenes.is_empty()) {
+		return 0;
+	}
 	return spawnable_scenes.size();
 }
```
Comment (nathanfranke) [inline scene/multiplayer/multiplayer_spawner.cpp:107]: Looking at the source code of LocalVector, I don't see why this is needed. `is_empty()` (inline) checks if `count == 0`, while `size()` (inline) returns `count`.

**Review by nathanfranke — CHANGES_REQUESTED** [review]

File: scene/multiplayer/multiplayer_spawner.cpp (hunk 1 above)
Comment (fire) [reply scene/multiplayer/multiplayer_spawner.cpp:107]: It crashes though. I don't know why.

Comment (nathanfranke) [reply scene/multiplayer/multiplayer_spawner.cpp:107]: I can't reproduce the crash at all with `get_spawnable_scene_count()` (Arch on 5.15.41-1-lts). I can reproduce the crash with the other method though. See my comment there.

File: scene/multiplayer/multiplayer_spawner.cpp [hunk 2]
```diff
@@ -97,12 +98,21 @@ void MultiplayerSpawner::add_spawnable_scene(const String &p_path) {
...
+
 String MultiplayerSpawner::get_spawnable_scene(int p_idx) const {
+	if (spawnable_scenes.is_empty()) {
+		return String();
+	}
 	return spawnable_scenes[p_idx].path;
 }
```
Comment (nathanfranke) [inline scene/multiplayer/multiplayer_spawner.cpp:114]: It being empty isn't the sole cause of the crash, it's just an out of bounds access. 

```suggestion
String MultiplayerSpawner::get_spawnable_scene(int p_idx) const {
	ERR_FAIL_COND_V(p_idx < 0, "");
	ERR_FAIL_COND_V(p_idx >= spawnable_scenes.size(), "");
	return spawnable_scenes[p_idx].path;
}
```

**Review by nathanfranke — CHANGES_REQUESTED** [review]

File: scene/multiplayer/multiplayer_spawner.cpp (hunk 2 above)
Comment (kleonc) [reply scene/multiplayer/multiplayer_spawner.cpp:114]: There are dedicated error macros for index checking:
```suggestion
String MultiplayerSpawner::get_spawnable_scene(int p_idx) const {
	ERR_FAIL_INDEX_V(p_idx, spawnable_scenes.size(), "");
	return spawnable_scenes[p_idx].path;
}
```

Comment (fire) [reply scene/multiplayer/multiplayer_spawner.cpp:114]: This fails with int32 is not uint32.

Comment (nathanfranke) [reply scene/multiplayer/multiplayer_spawner.cpp:114]: I think you just need to cast the size parameter. You'll lose the last bit though.

Comment (fire) [reply scene/multiplayer/multiplayer_spawner.cpp:114]: There's an uint version index check.

**Comment by fire** [comment]
I wasn't able to provide a test cast, is salvagable.

**Pull request closed without merging**